packages/server/src/middleware/nuxt.js

```diff
@@ -52,7 +52,7 @@ export default ({ options, nuxt, renderRoute, resources }) => async function nux
 
       const links = pushAssets
         ? pushAssets(req, res, publicPath, preloadFiles)
-        : defaultPushAssets(preloadFiles, shouldPush, publicPath, options.dev)
+        : defaultPushAssets(preloadFiles, shouldPush, publicPath, options)
```

Comment:
This change would be a breaking change for `2.x` releases (changing function signature). If the options than `dev` we can add it as another param. For `3.x` i suggest changing signature to accept an object with named args.

Reply:
`defaultPushAssets` is not the one exposed to users, it's only for internal use.
`pushAssets` is for user config, we can add options to it in 3.x.